pkg/controller/syncconfig/handler.go

```diff
@@ -2,12 +2,105 @@ package syncconfig
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
 func (r *ReconcileSyncConfig) handle(ctx context.Context, syncConfig *v1alpha1.SyncConfig, request reconcile.Request) error {
+	namespaces, err := r.getNamespaces(ctx, syncConfig, request)
+	if err != nil {
+		log.Error(err, "Error getting namespaces")
+		return err
+	}
+	var errors []error
+	for _, targetNamespace := range namespaces {
+		for _, deleteItem := range syncConfig.Spec.DeleteItems {
+			deleteObj := &unstructured.Unstructured{}
+			deleteObj.SetAPIVersion(deleteItem.APIVersion)
+			deleteObj.SetKind(deleteItem.Kind)
+			deleteObj.SetName(deleteItem.Name)
+			deleteObj.SetNamespace(targetNamespace.Name)
+
+			err = r.client.Delete(ctx, deleteObj, client.DeleteOptionFunc(func(d *client.DeleteOptions) {
+				option := metav1.DeletePropagationBackground
+				d.PropagationPolicy = &option
+			}))
+			if err != nil {
+				if !apierrors.IsNotFound(err) {
+					log.Error(err, "Error deleting object", getFullName(deleteObj)...)
+					errors = append(errors, err)
+				}
+			} else {
+				log.Info("Deleted", getFullName(deleteObj)...)
+			}
+		}
+		for _, item := range syncConfig.Spec.Items {
+			unstructObj := item.DeepCopy()
+			unstructObj.SetNamespace(targetNamespace.Name)
+
+			err := processTemplate(unstructObj, targetNamespace.Name)
+			if err != nil {
+				log.Error(err, "Error processing template", getFullName(unstructObj)...)
+				errors = append(errors, err)
+				continue
+			}
+
+			err = r.client.Create(ctx, unstructObj)
+			if err != nil {
+				if !apierrors.IsAlreadyExists(err) {
+					log.Error(err, "Error creating", getFullName(unstructObj)...)
+					errors = append(errors, err)
+					continue
+				}
+				existingObj := &unstructured.Unstructured{}
+				existingObj.SetKind(unstructObj.GetKind())
+				existingObj.SetAPIVersion(unstructObj.GetAPIVersion())
+				err = r.client.Get(ctx,
+					types.NamespacedName{
+						Name:      unstructObj.GetName(),
+						Namespace: unstructObj.GetNamespace()},
+					existingObj)
+				if err != nil {
+					errors = append(errors, err)
+					continue
+				}
+				unstructObj.SetResourceVersion(existingObj.GetResourceVersion())
+				err = r.client.Update(ctx, unstructObj)
+				if err != nil {
+					if apierrors.IsInvalid(err) && syncConfig.Spec.ForceRecreate {
+						log.Info("Recreating", getFullName(unstructObj)...)
+						err = recreateOject(ctx, unstructObj, r.client)
+						if err != nil {
+							log.Error(err, "Error recreating", getFullName(unstructObj)...)
+							errors = append(errors, err)
+						}
+					} else {
+						log.Error(err, "Error updating", getFullName(unstructObj)...)
+						errors = append(errors, err)
+					}
+				} else {
+					log.Info("Updated", getFullName(unstructObj)...)
+				}
+			} else {
+				log.Info("Created", getFullName(unstructObj)...)
+			}
+		}
+	}
+	if len(errors) > 0 {
```

Comment:
Use a custom error object with an `Append(error)` function and the formatting of the finial error message within its `String()` function.